src/config.rs

```diff
@@ -51,7 +51,16 @@ impl TryFrom<String> for Config {
 }
 
 impl Config {
-    pub(crate) fn new<P: AsRef<Path>>(path: P) -> Result<Option<Self>> {
+    pub(crate) fn from_path<P: AsRef<Path>>(path: P) -> Result<Option<Self>> {
+        let path = path.as_ref();
+
+        if !path.exists() {
```

Comment:
question: can you please elaborate why this block there with going to parent folders is required?

Reply:
If the folder specified contains the template config it should be used from there, but in case it doesn't its very nice that it picks up the config in the parent folder.

Somewhat akin to how it works with git config files etc.

It can be argued though, that the folder traversal could/should stop at the git repository root path.

Reply:
I see that this might be very useful. So I suggest:

1. this smart config finding logic is documented
2. we need here some mechanism to prevent a traversal beyond the actual template folder, so as you say stopping at the template root